src/game/path.service.js

```diff
@@ -58,28 +58,39 @@ angular.module('ngGo.Game.Path.Service', [
       return;
     }
 
+    //Decrement move
+    this.move--;
+
     //Delete path choice
-    if (this.path[this.move]) {
+    if (typeof this.path[this.move] !== 'undefined') {
+      var branch = this.path[this.move];
       delete this.path[this.move];
-      this.branches--;
+      if (branch > 0) {
+        this.branches--;
+      }
     }
 
-    //Decrement move
-    this.move--;
     return this;
```

Comment:
Sorry I put my comment in the wrong place, I was referring to the changes in lines 42-44 mostly.
I think the way the logic worked, was that it would only create an entry in  `this.path` for branches, not for the main path, and then delete it only if it was `> 0`. Was it an issue somewhere that we didn't set `this.path` value for the main branch?